src/core/lib/security/security_connector/s2a/s2a_security_connector.cc

```diff
@@ -44,46 +57,52 @@ class grpc_s2a_channel_security_connector final
       grpc_core::RefCountedPtr<grpc_channel_credentials> channel_creds,
       grpc_core::RefCountedPtr<grpc_call_credentials> request_metadata_creds,
       const char* target_name)
-      : grpc_channel_security_connector(/** url_scheme= **/ nullptr,
+      : grpc_channel_security_connector(/* url_scheme=*/nullptr,
                                         std::move(channel_creds),
                                         std::move(request_metadata_creds)),
-        target_name_(gpr_strdup(target_name)) {
-    // TODO(mattstev): implement.
-  }
+        target_name_(gpr_strdup(target_name)) {}
 
   ~grpc_s2a_channel_security_connector() override { gpr_free(target_name_); }
 
   void add_handshakers(
       const grpc_channel_args* args, grpc_pollset_set* interested_parties,
       grpc_core::HandshakeManager* handshake_manager) override {
-    // TODO(mattstev): implement.
+    // TODO(mattstev): implement. This is blocked by the implementation of the
+    // S2A TSI handshaker.
     return;
   }
 
   void check_peer(tsi_peer peer, grpc_endpoint* ep,
                   grpc_core::RefCountedPtr<grpc_auth_context>* auth_context,
                   grpc_closure* on_peer_checked) override {
-    // TODO(mattstev): implement.
-    return;
+    s2a_check_peer(peer, auth_context, on_peer_checked);
   }
 
   int cmp(const grpc_security_connector* other_sc) const override {
-    // TODO(mattstev): implement.
-    return 0;
+    auto* other =
+        reinterpret_cast<const grpc_s2a_channel_security_connector*>(other_sc);
+    int c = channel_security_connector_cmp(other);
+    if (c != 0) {
+      return c;
+    }
+    return strcmp(target_name_, other->target_name_);
   }
 
   bool check_call_host(grpc_core::StringView host,
```

Comment:
Do we want to do hostname verification check here?

Reply:
Per our offline discussion yesterday, I've removed this.